app/scripts/views/sign_up.js

```diff
@@ -212,57 +218,34 @@ function (Cocktail, _, p, BaseView, FormView, Template, AuthErrors, mailcheck,
 
     submit: function () {
       var self = this;
+      var experiment = self._experiments;
+
       return p()
         .then(function () {
           if (! self._isUserOldEnough()) {
+            experiment.saveState('coppaView', 'tooyoung');
+
             return self._cannotCreateAccount();
           }
 
-          self._wasMailcheckUseful();
+          experiment.signal('mailcheck', 'checkUseful', {
+            el: self.$el.find('.email')
+          });
+
+          experiment.saveState('coppaView', 'submit');
 
           return self._initAccount();
         });
     },
 
-    _isMailcheckEnabledValue: undefined,
-    _isMailcheckEnabled: function () {
-      // only check whether mailcheck is enabled once. Otherwise,
-      // an event is added to the able log every time the user
-      // blurs the email field, which could be multiple times.
-      if (typeof this._isMailcheckEnabledValue === 'undefined') {
-        var abData = {
-          isMetricsEnabledValue: this.metrics.isCollectionEnabled(),
-          uniqueUserId: this.user.get('uniqueUserId'),
-          // the window parameter will override any ab testing features
-          forceMailcheck: Url.searchParam('mailcheck', this.window.location.search)
-        };
-
-        this._isMailcheckEnabledValue =
-              this._able.choose('mailcheckEnabled', abData);
-      }
-      return this._isMailcheckEnabledValue;
-    },
-    _wasMailcheckUseful: function () {
-      var email = this.$el.find('.email');
-
-      if (email) {
-        var emailValue = email.val();
-        var mailcheckValue = email.data('mailcheckValue');
-        if (emailValue.length > 0 && mailcheckValue === emailValue) {
-          this.logScreenEvent('mailcheck-useful');
-        }
-      }
-    },
-
     onPasswordBlur: function () {
       var password = this.getElementValue('.password');
       this.checkPasswordStrength(password);
     },
 
-    suggestEmail: function () {
-      if (this._isMailcheckEnabled()) {
-        mailcheck(this.$el.find('.email'), this.metrics, this.translator);
-      }
+    onEmailBlur: function () {
+      this._experiments.saveState('mailcheck', 'enabled');
+      mailcheck(this.$el.find('.email'), this.metrics, this.translator, this._experiments);
```

Comment:
Instead of passing in this._experiments to mailcheck, can [these this line and the following line](https://github.com/mozilla/fxa-content-server/pull/2943/files#diff-15792598f0046820c70bf702c45f707fR57) be moved here?